### ISSUE TITLE
ID生成器的训练，为什么不用softmax直接回传梯度

### ISSUE THREAD

**Issue by zhuantouer** [agiresearch/IDGenRec#5]
hi，idea很好的文章

请教个问题，论文2.5.2，ID生成器也是个文本到文本的生成模型，生成的也是文本(token)，我理解和基础推荐模型框架是一致的。计算出logits后为什么不作softmax，然后就可以用交叉熵损失函数回传梯度了？比如生成"zeppelin cocktail bars tap"这个ID时，生成了zeppelin cocktail bars后，下个token logits softmax后和tab计算交叉熵是不是也能训练？

**Comment by evison** [comment]
Dear @zhuantouer, thanks for your interest in this work. The main reason is the token embedding has to be mapped to a specific discrete token, which breaks the gradient during back-propagation if we train the recommendation llm and id generation llm together. Our work provided one way to solve the problem, which is the alternative optimization method, but I do think this problem can be solved in other ways or even better ways. I think your idea of using cross-entropy to do co-training is worthy trying.
